diff -uNr netmaumau-0.23.2.orig/configure.ac netmaumau-0.23.2/configure.ac
--- netmaumau-0.23.2.orig/configure.ac	2015-07-18 09:51:32.000000000 +0200
+++ netmaumau-0.23.2/configure.ac	2015-07-18 12:01:41.238572673 +0200
@@ -7,7 +7,7 @@
 AM_INIT_AUTOMAKE([foreign dist-xz no-dist-gzip tar-pax no-installinfo no-texinfo.tex])
 AM_MAINTAINER_MODE
 AC_CONFIG_HEADERS([config.h])
-LT_INIT
+LT_INIT([disable-shared])
 LT_LANG([Windows Resource])
 
 AC_SUBST([BUILD_YEAR], [`date +%Y`])
@@ -143,6 +143,7 @@
 .mso www.tmac
 .URL \"http://www.gnu.org/software/gsl/manual/html_node/Random-number-environment-variables.html\" \"GSL Manual\" \" for more information\"" > gsl.h2m ],
   [ AM_CONDITIONAL([GSL], [ false ])] )
+
 PKG_CHECK_MODULES([LIBSQLITE], [sqlite3 >= 3.4.2])
 PKG_CHECK_MODULES([LIBLUA], [lua5.1 >= 5.1.2], [ AC_SUBST([PKGC_LUA], [lua5.1]) ],
   [ PKG_CHECK_MODULES([LIBLUA], [lua51 >= 5.1.2], [ AC_SUBST([PKGC_LUA], [lua51]) ],
@@ -252,7 +253,7 @@
 
 NMM_CLIENTVERSION([0],[15])
 
-AC_DEFINE([MAXPICBYTES], [1048576], [maximum size of player image in bytes (base64 encoded)])
+AC_DEFINE([MAXPICBYTES], [524288], [maximum size of player image in bytes (base64 encoded)])
 
 AC_DEFINE_UNQUOTED([SERVER_VERSION_MAJOR], [${SERVER_VERSION_MAJOR}], [server version major])
 AC_DEFINE_UNQUOTED([SERVER_VERSION_MINOR], [${SERVER_VERSION_MINOR}], [server version minor])
diff -uNr netmaumau-0.23.2.orig/src/client/Makefile.am netmaumau-0.23.2/src/client/Makefile.am
--- netmaumau-0.23.2.orig/src/client/Makefile.am	2015-06-19 13:03:15.000000000 +0200
+++ netmaumau-0.23.2/src/client/Makefile.am	2015-07-18 12:01:41.248572608 +0200
@@ -17,7 +17,7 @@
 	lostconnectionexception.cpp nonetmaumauserverexception.cpp playerlistexception.cpp \
 	protocolerrorexception.cpp scoresexception.cpp shutdownexception.cpp timeoutexception.cpp \
 	versionmismatchexception.cpp 
-libnetmaumauclient_la_LDFLAGS = -nodefaultlibs -nostartfiles -no-undefined \
+libnetmaumauclient_la_LDFLAGS = -mconsole -static-libgcc -static-libstdc++ -no-undefined \
 	-version-info 4:$(SERVER_VERSION_MINOR):$(SERVER_VERSION_MAJOR)
 libnetmaumauclient_la_LIBADD = libnetmaumauclient_private.la ../common/libnetmaumaucommon.la
 
diff -uNr netmaumau-0.23.2.orig/src/common/Makefile.am netmaumau-0.23.2/src/common/Makefile.am
--- netmaumau-0.23.2.orig/src/common/Makefile.am	2015-07-17 06:58:21.000000000 +0200
+++ netmaumau-0.23.2/src/common/Makefile.am	2015-07-18 12:01:41.248572608 +0200
@@ -32,6 +32,6 @@
 libnetmaumaucommon_la_CXXFLAGS = -I$(top_srcdir)/src/include 
 libnetmaumaucommon_la_SOURCES = abstractconnection.cpp abstractconnectionimpl.cpp \
 	abstractsocket.cpp abstractsocketimpl.cpp base64.cpp logger.cpp socketexception.cpp
-libnetmaumaucommon_la_LIBADD = libnetmaumaucommon_private.la
-libnetmaumaucommon_la_LDFLAGS = -nodefaultlibs -nostartfiles \
+libnetmaumaucommon_la_LIBADD = libnetmaumaucommon_private.la -lws2_32
+libnetmaumaucommon_la_LDFLAGS = -mconsole -static-libgcc -static-libstdc++ -no-undefined \
 	-version-info 5:$(SERVER_VERSION_MINOR):$(SERVER_VERSION_MAJOR)
diff -uNr netmaumau-0.23.2.orig/src/lua/Makefile.am netmaumau-0.23.2/src/lua/Makefile.am
--- netmaumau-0.23.2.orig/src/lua/Makefile.am	2015-04-29 13:02:05.000000000 +0200
+++ netmaumau-0.23.2/src/lua/Makefile.am	2015-07-18 12:01:41.248572608 +0200
@@ -16,4 +16,5 @@
 libluaruleset_la_CXXFLAGS = -I$(top_srcdir)/src/include -I$(top_srcdir)/src/common \
 	-I$(top_srcdir)/src/engine $(LIBLUA_CFLAGS)
 libluaruleset_la_SOURCES = luaexception.cpp luafatalexception.cpp luaruleset.cpp luastate.cpp
+libluaruleset_la_LDFLAGS = -static-libgcc -static-libstdc++ -no-undefined
 libluaruleset_la_LIBADD = $(LIBLUA_LIBS)
diff -uNr netmaumau-0.23.2.orig/src/server/Makefile.am netmaumau-0.23.2/src/server/Makefile.am
--- netmaumau-0.23.2.orig/src/server/Makefile.am	2015-07-17 06:58:21.000000000 +0200
+++ netmaumau-0.23.2/src/server/Makefile.am	2015-07-18 12:00:45.678929703 +0200
@@ -45,7 +45,6 @@
 libhttpd_la_CXXFLAGS = $(NO_EXCEPTIONS) $(LIBMICROHTTPD_CFLAGS) -I$(top_srcdir)/src/include \
 	-I$(top_srcdir)/src/common -I$(top_srcdir)/src/sqlite -I$(top_srcdir)/src/engine
 libhttpd_la_SOURCES  = httpd.cpp
-libhttpd_la_LIBADD   = $(LIBMICROHTTPD_LIBS)
 endif
 
 nmm_server_CPPFLAGS += -DPKGDATADIR="\"$(pkgdatadir)\""
@@ -54,8 +53,10 @@
 	-I$(top_srcdir)/src/lua $(POPT_CFLAGS)
 nmm_server_SOURCES = game.cpp main.cpp serverconnection.cpp servereventhandler.cpp \
 	serverplayer.cpp serverplayerexception.cpp
+nmm_server_LDFLAGS = -mconsole -all-static
 nmm_server_LDADD = ../common/libnetmaumaucommon.la libnmm_server_private.la \
-	../engine/libengine.la $(LMHL) $(POPT_LIBS) $(RT_LIBS)
+	../engine/libengine.la $(LMHL) $(LIBMICROHTTPD_LIBS) -lws2_32 $(POPT_LIBS) \
+	$(RT_LIBS)
 
 if RC
 nmm_server_LDADD += netmaumau.rc.lo
diff -uNr netmaumau-0.23.2.orig/src/sqlite/Makefile.am netmaumau-0.23.2/src/sqlite/Makefile.am
--- netmaumau-0.23.2.orig/src/sqlite/Makefile.am	2015-04-30 21:32:58.000000000 +0200
+++ netmaumau-0.23.2/src/sqlite/Makefile.am	2015-07-18 12:01:52.398497924 +0200
@@ -6,6 +6,7 @@
 	-DDBDIR=\"$(localstatedir)/$(PACKAGE)\" -DBUILDDIR=\"$(abs_top_builddir)\"
 libsqlite_la_CXXFLAGS = $(LIBSQLITE_CFLAGS) $(NO_EXCEPTIONS)
 libsqlite_la_SOURCES = sqlite.cpp sqliteimpl.cpp
+libsqlite_la_LDFLAGS = -static-libgcc -static-libstdc++ -no-undefined
 libsqlite_la_LIBADD = $(LIBSQLITE_LIBS)
 
 install-data-local:
diff -uNr netmaumau-0.23.2.orig/src/test/Makefile.am netmaumau-0.23.2/src/test/Makefile.am
--- netmaumau-0.23.2.orig/src/test/Makefile.am	2015-06-16 20:25:40.000000000 +0200
+++ netmaumau-0.23.2/src/test/Makefile.am	2015-07-18 12:01:52.398497924 +0200
@@ -33,7 +33,7 @@
 	-I$(top_srcdir)/src/sqlite
 test_netmaumau_SOURCES = test_netmaumau.cpp testeventhandler.cpp
 test_netmaumau_LDADD = ../common/libnetmaumaucommon.la ../engine/libengine.la
-test_netmaumau_LDFLAGS = -no-install
+test_netmaumau_LDFLAGS = -all-static
 
 if ENABLE_CLI_CLIENT
 nmm_client_CPPFLAGS = -DCLIENTVERSION=$(CLIENTVERSION) $(GSL)
