diff -uNr netmaumau-0.8.orig/configure.ac netmaumau-0.8/configure.ac
--- netmaumau-0.8.orig/configure.ac	2015-01-02 11:54:02.000000000 +0100
+++ netmaumau-0.8/configure.ac	2015-01-02 12:00:51.605090632 +0100
@@ -7,7 +7,7 @@
 AM_INIT_AUTOMAKE([foreign dist-xz no-dist-gzip tar-pax no-installinfo no-texinfo.tex])
 AM_MAINTAINER_MODE
 AC_CONFIG_HEADERS([config.h])
-LT_INIT
+LT_INIT([disable-shared])
 LT_LANG([Windows Resource])
 
 AC_SUBST([BUILD_YEAR], [`date +%Y`])
@@ -140,7 +140,7 @@
 AC_SUBST([SERVER_VERSION_MAJOR])
 AC_SUBST([SERVER_VERSION_MINOR])
 
-AC_DEFINE([MAXPICBYTES], [1048576], [maximum size of player image in bytes (base64 encoded)])
+AC_DEFINE([MAXPICBYTES], [524288], [maximum size of player image in bytes (base64 encoded)])
 
 AC_DEFINE_UNQUOTED([SERVER_VERSION_MAJOR], [${SERVER_VERSION_MAJOR}], [server version major])
 AC_DEFINE_UNQUOTED([SERVER_VERSION_MINOR], [${SERVER_VERSION_MINOR}], [server version minor])
diff -uNr netmaumau-0.8.orig/src/client/Makefile.am netmaumau-0.8/src/client/Makefile.am
--- netmaumau-0.8.orig/src/client/Makefile.am	2014-12-29 07:57:58.000000000 +0100
+++ netmaumau-0.8/src/client/Makefile.am	2015-01-02 12:00:51.615090391 +0100
@@ -16,7 +16,7 @@
 	connectionrejectedexception.cpp interceptederrorexception.cpp \
 	nonetmaumauserverexception.cpp playerlistexception.cpp protocolerrorexception.cpp \
 	shutdownexception.cpp timeoutexception.cpp versionmismatchexception.cpp 
-libnetmaumauclient_la_LDFLAGS = -nodefaultlibs -nostartfiles -no-undefined \
+libnetmaumauclient_la_LDFLAGS = -static-libgcc -static-libstdc++ -no-undefined \
 	-version-info 3:$(SERVER_VERSION_MINOR):$(SERVER_VERSION_MAJOR)
 libnetmaumauclient_la_LIBADD = libnetmaumauclient_private.la ../common/libnetmaumaucommon.la
 
diff -uNr netmaumau-0.8.orig/src/common/Makefile.am netmaumau-0.8/src/common/Makefile.am
--- netmaumau-0.8.orig/src/common/Makefile.am	2014-12-29 07:57:59.000000000 +0100
+++ netmaumau-0.8/src/common/Makefile.am	2015-01-02 12:00:51.625090151 +0100
@@ -20,6 +20,6 @@
 libnetmaumaucommon_la_CXXFLAGS = -I$(top_srcdir)/src/include
 libnetmaumaucommon_la_SOURCES = abstractconnection.cpp abstractconnectionimpl.cpp \
 	abstractsocket.cpp abstractsocketimpl.cpp base64.cpp logger.cpp socketexception.cpp
-libnetmaumaucommon_la_LIBADD = libnetmaumaucommon_private.la
-libnetmaumaucommon_la_LDFLAGS = -nodefaultlibs -nostartfiles \
+libnetmaumaucommon_la_LIBADD = libnetmaumaucommon_private.la -lws2_32
+libnetmaumaucommon_la_LDFLAGS = -static-libgcc -static-libstdc++ -no-undefined \
 	-version-info 1:$(SERVER_VERSION_MINOR):$(SERVER_VERSION_MAJOR)
diff -uNr netmaumau-0.8.orig/src/server/Makefile.am netmaumau-0.8/src/server/Makefile.am
--- netmaumau-0.8.orig/src/server/Makefile.am	2015-01-02 11:54:02.000000000 +0100
+++ netmaumau-0.8/src/server/Makefile.am	2015-01-02 12:00:51.625090151 +0100
@@ -37,6 +37,7 @@
 
 nmm_server_CXXFLAGS = -I$(top_srcdir)/src/include -I$(top_srcdir)/src/common \
 	-I$(top_srcdir)/src/engine $(POPT_CFLAGS)
+nmm_server_LDFLAGS = -all-static
 nmm_server_SOURCES = game.cpp main.cpp serverconnection.cpp servereventhandler.cpp \
 	serverplayer.cpp serverplayerexception.cpp
 nmm_server_LDADD = ../common/libnetmaumaucommon.la ../engine/libengine.la $(POPT_LIBS)
diff -uNr netmaumau-0.8.orig/src/test/Makefile.am netmaumau-0.8/src/test/Makefile.am
--- netmaumau-0.8.orig/src/test/Makefile.am	2014-12-26 05:16:22.000000000 +0100
+++ netmaumau-0.8/src/test/Makefile.am	2015-01-02 12:00:51.625090151 +0100
@@ -28,7 +28,7 @@
 	-I$(top_srcdir)/src/engine
 test_netmaumau_SOURCES = test_netmaumau.cpp testeventhandler.cpp
 test_netmaumau_LDADD = ../common/libnetmaumaucommon.la ../engine/libengine.la
-test_netmaumau_LDFLAGS = -no-install
+test_netmaumau_LDFLAGS = -all-static
 
 if ENABLE_CLI_CLIENT
 nmm_client_CXXFLAGS = -I$(top_srcdir)/src/include -I$(top_srcdir)/src/common \
