diff -uNr netmaumau-0.6.orig/configure.ac netmaumau-0.6/configure.ac
--- netmaumau-0.6.orig/configure.ac	2014-12-15 10:44:12.000000000 +0100
+++ netmaumau-0.6/configure.ac	2014-12-15 11:23:21.765638803 +0100
@@ -7,7 +7,7 @@
 AM_INIT_AUTOMAKE([foreign])
 AM_MAINTAINER_MODE
 AC_CONFIG_HEADERS([config.h])
-LT_INIT
+LT_INIT([disable-shared])
 LT_LANG([Windows Resource])
 
 AC_DEFINE_UNQUOTED([BUILD_DATE], [`date +%s`], [the build date in Unix time])
@@ -111,7 +111,7 @@
 AC_SUBST([SERVER_VERSION_MAJOR])
 AC_SUBST([SERVER_VERSION_MINOR])
 
-AC_DEFINE([MAXPICBYTES], [1048576], [maximum size of player image in bytes (base64 encoded)])
+AC_DEFINE([MAXPICBYTES], [524288], [maximum size of player image in bytes (base64 encoded)])
 
 AC_DEFINE_UNQUOTED([SERVER_VERSION_MAJOR], [${SERVER_VERSION_MAJOR}], [server version major])
 AC_DEFINE_UNQUOTED([SERVER_VERSION_MINOR], [${SERVER_VERSION_MINOR}], [server version minor])
diff -uNr netmaumau-0.6.orig/src/client/Makefile.am netmaumau-0.6/src/client/Makefile.am
--- netmaumau-0.6.orig/src/client/Makefile.am	2014-12-08 06:32:46.000000000 +0100
+++ netmaumau-0.6/src/client/Makefile.am	2014-12-15 11:23:21.765638803 +0100
@@ -9,7 +9,7 @@
 	interceptederrorexception.cpp timeoutexception.cpp shutdownexception.cpp \
 	playerlistexception.cpp capabilitiesexception.cpp protocolerrorexception.cpp \
 	connectionrejectedexception.cpp versionmismatchexception.cpp nonetmaumauserverexception.cpp
-libnetmaumauclient_la_LDFLAGS = -nodefaultlibs -nostartfiles -no-undefined \
+libnetmaumauclient_la_LDFLAGS = -static-libgcc -static-libstdc++ -no-undefined \
 	-version-info 1:$(SERVER_VERSION_MINOR):$(SERVER_VERSION_MAJOR)
 libnetmaumauclient_la_LIBADD = ../common/libnetmaumaucommon.la
 
diff -uNr netmaumau-0.6.orig/src/common/Makefile.am netmaumau-0.6/src/common/Makefile.am
--- netmaumau-0.6.orig/src/common/Makefile.am	2014-12-15 10:44:12.000000000 +0100
+++ netmaumau-0.6/src/common/Makefile.am	2014-12-15 11:23:21.775639823 +0100
@@ -7,5 +7,6 @@
 libnetmaumaucommon_la_CXXFLAGS = -I$(top_srcdir)/src/include
 libnetmaumaucommon_la_SOURCES = abstractsocket.cpp abstractconnection.cpp socketexception.cpp \
 	cardtools.cpp logger.cpp base64.cpp errorstring.cpp pngcheck.cpp
-libnetmaumaucommon_la_LDFLAGS = -nodefaultlibs -nostartfiles \
+libnetmaumaucommon_la_LDFLAGS = -static-libgcc -static-libstd++ -no-undefined \
 	-version-info 0:$(SERVER_VERSION_MINOR):$(SERVER_VERSION_MAJOR)
+libnetmaumaucommon_la_LIBADD = -lws2_32
diff -uNr netmaumau-0.6.orig/src/server/Makefile.am netmaumau-0.6/src/server/Makefile.am
--- netmaumau-0.6.orig/src/server/Makefile.am	2014-12-13 09:50:19.000000000 +0100
+++ netmaumau-0.6/src/server/Makefile.am	2014-12-15 11:23:21.775639823 +0100
@@ -23,6 +23,7 @@
 
 nmm_server_CXXFLAGS = -I$(top_srcdir)/src/include -I$(top_srcdir)/src/common \
 	-I$(top_srcdir)/src/engine $(POPT_CFLAGS)
+nmm_server_LDFLAGS = -all-static
 nmm_server_SOURCES = main.cpp serverconnection.cpp serverplayer.cpp servereventhandler.cpp \
 	game.cpp serverplayerexception.cpp
 nmm_server_LDADD = ../common/libnetmaumaucommon.la ../engine/libengine.la $(POPT_LIBS)
diff -uNr netmaumau-0.6.orig/src/test/Makefile.am netmaumau-0.6/src/test/Makefile.am
--- netmaumau-0.6.orig/src/test/Makefile.am	2014-12-10 16:00:49.000000000 +0100
+++ netmaumau-0.6/src/test/Makefile.am	2014-12-15 11:23:21.775639823 +0100
@@ -11,13 +11,13 @@
 test_netmaumau_CXXFLAGS = -I$(top_srcdir)/src/include -I$(top_srcdir)/src/common \
 	-I$(top_srcdir)/src/engine
 test_netmaumau_SOURCES = test_netmaumau.cpp testeventhandler.cpp
+test_netmaumau_LDFLAGS = -all-static
 test_netmaumau_LDADD = ../common/libnetmaumaucommon.la ../engine/libengine.la
-test_netmaumau_LDFLAGS = -no-install
 
 if ENABLE_CLIENT
 test_client_CXXFLAGS = -I$(top_srcdir)/src/include -I$(top_srcdir)/src/common \
 	-I$(top_srcdir)/src/client $(POPT_CFLAGS)
 test_client_SOURCES = test_client.cpp testclient.cpp
+test_client_LDFLAGS = -all-static
 test_client_LDADD = ../client/libnetmaumauclient.la ../common/libnetmaumaucommon.la $(POPT_LIBS)
-test_client_LDFLAGS = -no-install
 endif
